From 687fc76f6b56f0cb4dc18218025d23f24a87900f Mon Sep 17 00:00:00 2001
Message-Id: <687fc76f6b56f0cb4dc18218025d23f24a87900f.1431284209.git.chang-joon.lee@intel.com>
In-Reply-To: <89460f3fc3d6c0659d44897a6adec3c3fecc5564.1431284209.git.chang-joon.lee@intel.com>
References: <89460f3fc3d6c0659d44897a6adec3c3fecc5564.1431284209.git.chang-joon.lee@intel.com>
From: Michel Thierry <michel.thierry@intel.com>
Date: Fri, 10 Apr 2015 17:00:04 +0100
Subject: [PATCH 31/68] SQUASHME! [VPG]: drm/i915: Fix for gen7 systems with
 no ppgtt

DPA (yes, dynamic page alloc has now an acronym) introduced a regression
in BYT. BYT-Android doesn't use any PPGTT, only global GTT. This showed
a problem in the _initialized_ logic caused by one of the DPA patches.

This patch could be squashed in to the one that originally modified the
uninitialized variable:

commit b133cb6ce97f7fd051f1eb1c8efa438b98e1b92f
Author: Ben Widawsky <benjamin.widawsky@intel.com>
Date:   Mon Aug 18 21:40:52 2014 -0700

    FROM_UPSTREAM [VPG]: drm/i915: Initialize all contexts

Change-Id: I9e92013a5441736d0370ea18cb1fed8fdf3465c7
For: GMINL-5449
For: VIZ-4619
Signed-off-by: Michel Thierry <michel.thierry@intel.com>
---
 drivers/gpu/drm/i915/i915_gem_context.c |    6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/i915_gem_context.c b/drivers/gpu/drm/i915/i915_gem_context.c
index ba4b893..4f05fe8 100644
--- a/drivers/gpu/drm/i915/i915_gem_context.c
+++ b/drivers/gpu/drm/i915/i915_gem_context.c
@@ -819,7 +819,11 @@ static int do_switch(struct intel_engine_cs *ring,
 		i915_gem_context_unreference(from);
 	}
 
-	uninitialized = !to->legacy_hw_ctx.initialized;
+	if (INTEL_INFO(ring->dev)->gen < 8)
+		uninitialized = !to->legacy_hw_ctx.initialized && from == NULL;
+	else
+		uninitialized = !to->legacy_hw_ctx.initialized;
+
 	to->legacy_hw_ctx.initialized = true;
 
 done:
-- 
1.7.9.5

